core/lib/spree/core/role_configuration.rb

```diff
@@ -66,5 +66,16 @@ def assign_permissions role_name, permission_sets
       roles[name].permission_sets |= permission_sets
       roles[name]
     end
+
+    # Unassign permission sets for a {Spree::Role} that has the name of role_name
+    # @param role_name [Symbol, String] The name of the role to disassociate permissions with
+    # @param permission_sets [Array<Spree::PermissionSets::Base>, Set<Spree::PermissionSets::Base>]
+    #   A list of permission sets to remove from being activateable if the user has the role indicated by role_name
+    def unassign_permissions role_name, permission_sets
+      name = role_name.to_s
+
+      roles[name].permission_sets -= permission_sets
```

Comment:
You should use `Hash#fetch` here when you know the key is suposed to be present. It makes no sense to use `Hash#[]` that will in the error case return `nil` hiding the "true" source of the error when trying to call `#permission_sets` on `nil`.